src/net.cpp

```diff
@@ -2034,6 +2037,10 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                 addr = addrman.Select();
             }
 
+            // The following checks are never applied to MANUAL and ADDR_FETCH,
+            // because those connections are made elsewhere.
+            assert(conn_type != ConnectionType::MANUAL && conn_type != ConnectionType::ADDR_FETCH);
```

Comment:
The following two variants are equivalent:

```cpp
assert(A && B);
```
vs
```cpp
assert(A);
assert(B);
```

However if the first variant fails it would be impossible to say which one of `A` or `B` was false just by looking at the message (something like `assertion "A && B" failed`). With the second variant it is clear whether `A` or `B` was false.